core/string/ustring.cpp

```diff
@@ -88,6 +88,31 @@ bool select_word(const String &p_s, int p_col, int &r_beg, int &r_end) {
 	}
 }
 
+template <typename LHS, typename RHS>
+_FORCE_INLINE_ constexpr bool string_equal(const LHS *lhs, const RHS *rhs, size_t len) {
+	if constexpr (std::is_same_v<LHS, char32_t> && std::is_same_v<RHS, char32_t>) {
```

Comment:
`std::is_same_v<LHS, RHS>`  and then `memcmp(lhs, rhs, len * sizeof(LHS))` ? Or is it an error not to have `char32_t` involved ?